cpp/src/arrow/array.h

```diff
@@ -450,6 +450,9 @@ class ARROW_EXPORT ListArray : public Array {
   /// \brief Return array object containing the list's values
   std::shared_ptr<Array> values() const;
 
+  /// \brief Unnests the Array by one level
+  std::shared_ptr<Array> Flatten() const { return values(); }
```

Comment:
I'm not really sure what's the purpose of adding this redundant API. Really we need to implement a flatten _kernel_ but having "flatten" available in Python is useful. Suggest removing this

Reply:
https://issues.apache.org/jira/browse/ARROW-3520

Reply:
Ok, will remove.